ext/standard/file.c

```diff
@@ -543,7 +543,7 @@ PHP_FUNCTION(file_get_contents)
 		RETURN_FALSE;
 	}
 
-	if (offset > 0 && php_stream_seek(stream, offset, SEEK_SET) < 0) {
+	if (offset != 0 && php_stream_seek(stream, offset, ((offset > 0) ? SEEK_SET : SEEK_END)) < 0) {
```

Comment:
Yeah, doesn't look like the stream wrapper API has any reasonable way to support this, so I'm fine leaving this part out.